Sources/ETCD/ETCDClient.swift

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-etcd-client-gsoc open source project
+//
+// Copyright (c) 2024 Apple Inc. and the swift-etcd-client-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-etcd-client-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import GRPC
+import NIO
+import SwiftProtobuf
+
+public final class EtcdClient: @unchecked Sendable {
+    private let host: String
+    private let port: Int
+    private var group: EventLoopGroup
+    private var connection: ClientConnection
+    private var client: Etcdserverpb_KVNIOClient!
```

Comment:
We don't need a force unwrap here
```suggestion
    private var client: Etcdserverpb_KVNIOClient
```